src/lib/dateLib.ts

```diff
@@ -173,3 +178,206 @@ export const dateLib = {
   startOfWeek,
   startOfYear
 };
+
+/**
+ * The date library used by DayPicker. It's a subset of the date-fns functions
+ * with an explicit locale / formatting options.
+ *
+ * @private
+ * @internal
+ */
+export class DateLib {
```

Comment:
Would this work somehow?

```suggestion
export class DateLib extends DefaultDateLib {
```